Global mysqldump configuration ignored

## Bug Report

**Current Behavior**
Given `mysqldump` version 8 and a MySQL server version 5.7.

Dumping a TYPO3 DB using `database:export` fails because `mysqldump` tries to read the `COLUMN_STATISTICS` table which was added with MySQL server version 8.

A regular `mysqldump` works fine since `/etc/mysql/my.cnf` contains the following to disable the column statistics feature:

```ini
[mysqldump]
column-statistics=0
```

However, this is ignored by the `database:export` command since it internally uses `--defaults-file` to preset the DB credentials for access.

**Expected behavior/output**
The `database:export` command should succeeed.

**TYPO3 Console command executed**
```
typo3cms database:export > test.sql
```

**TYPO3 Console command output**
```
-- Connecting to mysql...
-- Starting transaction...
-- Setting savepoint...
-- Retrieving table structure for table backend_layout...
-- Sending SELECT query...
-- Retrieving rows...
mysqldump: Couldn't execute 'SELECT COLUMN_NAME,                       JSON_EXTRACT(HISTOGRAM, '$."number-of-buckets-specified"')                FROM information_schema.COLUMN_STATISTICS                WHERE SCHEMA_NAME = 'typo3' AND TABLE_NAME = 'backend_layout';': Unknown table 'COLUMN_STATISTICS' in information_schema (1109)

```

**Environment**
- TYPO3 Console version(s): 6.6.0
- TYPO3 version(s): 10.4.20
- PHP version(s): 7.4.22
- Is your TYPO3 installation set up with Composer (Composer Mode): yes
- OS: Docker (Ubuntu 20.04)

**Possible Solution**
Revert the change in https://github.com/TYPO3-Console/TYPO3-Console/pull/987 to switch back to `---defaults-extra-file` instead of `--defaults-file` to have `mysqldump` respect the configuration in `/etc/mysql/my.cnf` again:

```
--defaults-file=#       Only read default options from the given file #.
--defaults-extra-file=# Read this file after the global files are read.
```

Alternatively `column-statistics=0` could be added to the generated `mysqldump` configuration, but then older versions will choke:

```
-- Dump of TYPO3 Connection "Default"
mysqldump: [ERROR] unknown variable 'column-statistics=0'
Could not dump SQL for connection "Default",
```

**Additional context**
Server Fault: [mysqldump throws: Unknown table 'COLUMN_STATISTICS' in information_schema (1109)](https://serverfault.com/questions/912162/mysqldump-throws-unknown-table-column-statistics-in-information-schema-1109)

--- a/Classes/Console/Command/Database/DatabaseExportCommand.php
+++ b/Classes/Console/Command/Database/DatabaseExportCommand.php
@@ -18,6 +18,7 @@ use Helhum\Typo3Console\Database\Configuration\ConnectionConfiguration;
 use Helhum\Typo3Console\Database\Process\MysqlCommand;
 use Helhum\Typo3Console\Database\Schema\TableMatcher;
 use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Input\InputOption;
 use Symfony\Component\Console\Output\OutputInterface;
@@ -51,6 +52,7 @@ Tables to be excluded from the export can be specified fully qualified or with w
 <b>Example:</b>
 
   <code>%command.full_name% -c Default -e 'cf_*' -e 'cache_*' -e '[bf]e_sessions' -e sys_log</code>
+  <code>%command.full_name% database:export -c Default -- --column-statistics=0</code>
 EOH
         );
         $this->setDefinition([
@@ -68,11 +70,18 @@ EOH
                 'TYPO3 database connection name (defaults to all configured MySQL connections)',
                 null
             ),
+            new InputArgument(
+                'additionalMysqlDumpArguments',
+                InputArgument::IS_ARRAY,
+                'Pass one or more additional arguments to the mysqldump command; see examples',
+                []
+            ),
         ]);
     }
 
     protected function execute(InputInterface $input, OutputInterface $output)
     {
+        $additionalMysqlDumpArguments = $input->getArgument('additionalMysqlDumpArguments');
         $connection = $input->getOption('connection');
         $excludes = $input->getOption('exclude');
 
@@ -91,7 +100,7 @@ EOH
         foreach ($availableConnectionNames as $mysqlConnectionName) {
             $mysqlCommand = new MysqlCommand($this->connectionConfiguration->build($mysqlConnectionName), $output);
             $exitCode = $mysqlCommand->mysqldump(
-                $this->buildArguments($mysqlConnectionName, $excludes, $output),
+                array_merge($this->buildArguments($mysqlConnectionName, $excludes, $output), $additionalMysqlDumpArguments),
                 null,
                 $mysqlConnectionName
             );

--- a/Documentation/CommandReference/CacheFlushtags.rst
+++ b/Documentation/CommandReference/CacheFlushtags.rst
@@ -18,7 +18,7 @@ Flushes caches by tags, optionally only caches in specified groups.
 **Example:**
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms cache:flushtags news_123 --groups pages,all
 

--- a/Documentation/CommandReference/ConfigurationRemove.rst
+++ b/Documentation/CommandReference/ConfigurationRemove.rst
@@ -21,7 +21,7 @@ LocalConfiguration.php and not be overridden elsewhere.
 **Example:**
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms configuration:remove DB,EXT/EXTCONF/realurl
 

--- a/Documentation/CommandReference/ConfigurationSet.rst
+++ b/Documentation/CommandReference/ConfigurationSet.rst
@@ -18,22 +18,22 @@ Set system configuration option value by path.
 **Examples:**
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms configuration:set SYS/fileCreateMask 0664
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms configuration:set EXTCONF/processor_enabled true --json
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms configuration:set EXTCONF/lang/availableLanguages '["de", "fr"]' --json
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms configuration:set configuration:set BE/adminOnly -- -1
 

--- a/Documentation/CommandReference/ConfigurationShow.rst
+++ b/Documentation/CommandReference/ConfigurationShow.rst
@@ -20,7 +20,7 @@ the difference between these values is shown.
 **Example:**
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms configuration:show DB
 

--- a/Documentation/CommandReference/ConfigurationShowactive.rst
+++ b/Documentation/CommandReference/ConfigurationShowactive.rst
@@ -19,7 +19,7 @@ Shows the configuration value that is currently effective, no matter where and h
 **Example:**
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms configuration:showactive DB --json
 

--- a/Documentation/CommandReference/ConfigurationShowlocal.rst
+++ b/Documentation/CommandReference/ConfigurationShowlocal.rst
@@ -20,7 +20,7 @@ Note that this value could be overridden. Use `typo3cms configuration:show <path
 **Example:**
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms configuration:showlocal DB
 

--- a/Documentation/CommandReference/DatabaseExport.rst
+++ b/Documentation/CommandReference/DatabaseExport.rst
@@ -22,7 +22,7 @@ Tables to be excluded from the export can be specified fully qualified or with w
 **Example:**
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms database:export -c Default -e 'cf_*' -e 'cache_*' -e '[bf]e_sessions' -e sys_log
 

--- a/Documentation/CommandReference/DatabaseExport.rst
+++ b/Documentation/CommandReference/DatabaseExport.rst
@@ -28,6 +28,18 @@ Tables to be excluded from the export can be specified fully qualified or with w
 
 
 
+.. code-block:: shell
+
+   typo3cms database:export database:export -c Default -- --column-statistics=0
+
+
+Arguments
+=========
+
+`additionalMysqlDumpArguments`
+   Pass one or more additional arguments to the mysqldump command; see examples
+
+
 
 Options
 =======

--- a/Documentation/CommandReference/DatabaseImport.rst
+++ b/Documentation/CommandReference/DatabaseImport.rst
@@ -21,21 +21,21 @@ This obviously only works when MySQL is used as DBMS.
 **Example (import):**
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    ssh remote.server '/path/to/typo3cms database:export' | typo3cms database:import
 
 **Example (select):**
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    echo 'SELECT username from be_users WHERE admin=1;' | typo3cms database:import
 
 **Example (interactive):**
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms database:import --interactive
 

--- a/Documentation/CommandReference/DatabaseUpdateschema.rst
+++ b/Documentation/CommandReference/DatabaseUpdateschema.rst
@@ -40,19 +40,19 @@ To avoid shell matching all types with wildcards should be quoted.
 **Example:**
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms database:updateschema "*.add,*.change"
 
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms database:updateschema "*.add" --raw
 
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms database:updateschema "*" --verbose
 

--- a/Documentation/CommandReference/Help.rst
+++ b/Documentation/CommandReference/Help.rst
@@ -16,14 +16,14 @@ help
 The `help` command displays help for a given command:
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms help list
 
 You can also output the help in other formats by using the **--format** option:
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms help --format=xml list
 

--- a/Documentation/CommandReference/InstallGeneratepackagestates.rst
+++ b/Documentation/CommandReference/InstallGeneratepackagestates.rst
@@ -26,7 +26,7 @@ Marks the following extensions as active:
 To require TYPO3 core extensions use the following command:
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    composer require typo3/cms-foo "*"
 
@@ -35,7 +35,7 @@ This updates your composer.json and composer.lock without any other changes.
 **Example:**
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms install:generatepackagestates
 

--- a/Documentation/CommandReference/List.rst
+++ b/Documentation/CommandReference/List.rst
@@ -16,28 +16,28 @@ list
 The `list` command lists all commands:
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms list
 
 You can also display the commands for a specific namespace:
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms list test
 
 You can also output the information in other formats by using the **--format** option:
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms list --format=xml
 
 It's also possible to get raw list of commands (useful for embedding command runner):
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms list --raw
 

--- a/Documentation/CommandReference/UpgradeRun.rst
+++ b/Documentation/CommandReference/UpgradeRun.rst
@@ -21,32 +21,32 @@ When no identifier is specified a select UI is presented to select a wizard out 
 **Examples:**
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms upgrade:run all
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms upgrade:run all --confirm all
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms upgrade:run argon2iPasswordHashes --confirm all
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms upgrade:run all --confirm all --deny typo3DbLegacyExtension --deny funcExtension
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms upgrade:run all --deny all
 
 
-.. code-block:: shell 
+.. code-block:: shell
 
    typo3cms upgrade:run all --no-interaction --deny all --confirm argon2iPasswordHashes
 

--- a/Packages/CreateReferenceCommand/src/Command/CommandReferenceRenderCommand.php
+++ b/Packages/CreateReferenceCommand/src/Command/CommandReferenceRenderCommand.php
@@ -233,7 +233,7 @@ class CommandReferenceRenderCommand extends \Symfony\Component\Console\Command\C
     {
         $output = $input;
         // replace multiline comments with a code-block
-        $output = preg_replace('/^(\s*\<code\>)(.*)(\<\/code\>\s*)$/m', "\n\n.. code-block:: shell \n\n   $2\n", $output);
+        $output = preg_replace('/^(\s*\<code\>)(.*)(\<\/code\>\s*)$/m', "\n\n.. code-block:: shell\n\n   $2\n", $output);
         $output = preg_replace('|\<b>(((?!\</b>).)*)\</b>|', '**$1**', $output);
         $output = preg_replace('|\<i>(((?!\</i>).)*)\</i>|', '*$1*', $output);
         $output = preg_replace('|\<u>(((?!\</u>).)*)\</u>|', '*$1*', $output);

